### PR TITLE
Fix for sigma_i going to zero in ITD landfast ice.

### PR DESCRIPTION
**Pull request by kshedstrom**
- We can't divide by zero, so don't let sigma_i be zero.

### CODE REVIEW

**Review by Hallberg-NOAA — COMMENTED** [review]
I do not understand why this change is necessary.   When I look at the code as a whole, I see that the expression for sigma_i about 20 lines above this change, `sigma_i = max(sqrt(log(1.0 + v_i/m_i**2)), CS%puny)` at line 1959 of this same file, already includes a max with `CS%puny`, and I am not seeing anythng between these lines that changes the value of `sigma_i`.  What am I missing here?

**Comment by kshedstrom** [comment]
Yes, you are right. I don't remember how I thought sigma_i was getting a zero, except that it was thin new ice.

**Comment by kshedstrom** [comment]
Now I remember - thin new ice lead to v_i being zero.

**Comment by kshedstrom** [comment]
Sorry, @Hallberg-NOAA, getting my branches confused. How about this one?

**Review by Hallberg-NOAA — APPROVED** [review]
This change makes much more sense than the previous version!

File: src/SIS_dyn_cgrid.F90 [hunk 1]
```diff
@@ -1952,14 +1953,14 @@ subroutine basal_stress_coeff_itd(G, IG, IST, sea_lev, CS)
         do n =1, ncat
           v_i = v_i + vcat(n)**2 / (max(acat(n), CS%puny))
         enddo
-        v_i = v_i - m_i**2
+        v_i = max( (v_i - m_i**2), CS%puny )
```
Comment (Hallberg-NOAA) [inline src/SIS_dyn_cgrid.F90:1956]: I think that this expression is not dimensionally consistent. CS%puny is described as being non-dimensional (and is used as such on line 1960, but here it is being used in a expression with units of [m2].  Should there be a separate CS%puny_area variable?

**Comment by marshallward** [comment]
@kshedstrom Could you do something like this?
```
if (sigma_i > 0) then
   g_k(:) = exp(-(log(x_k(:)/CS%onemeter) - mu_i) ** 2 / (2.0 * sigma_i ** 2)) / &
                 (x_k(:) * sigma_i * sqrt(2.0 * pi))
                 
   ...
   Tbt(:) = ...
else
   ! g_k(:) = 0., since exp(-a**2/n**2)/n -> 0 as n->0
   ! P_x(:) = 0.
   ! tb_tmp(:) = 0.

   Tbt(:) = 0.
endif
```
It would avoid the need for `CS%puny` here, and might also reduce the computation.

**Comment by marshallward** [comment]
Thanks for testing it out, let's see what happens :sweat_smile:.

However, I think the better test would be to remove `puny_m2`, rather than introducing it alongside the `sigma_i > 0` test.  `sigma_i` is only zero if `v_i` is zero, so if you bound `v_i` then I would assume that it's not even possible for `sigma_i` to be zero.

I was thinking that the `sigma_i > 0` test would avoid the need to introduce another "puny" parameter, but I suppose that either approach gets you to the same place.

**Comment by kshedstrom** [comment]
Try that one?

**Comment by marshallward** [comment]
That's what I had in mind, I guess the question is whether it fixed your problem without creating any new ones?

**Comment by kshedstrom** [comment]
My short job ran fine, a longer one is stuck in the queue.

**Comment by kshedstrom** [comment]
Yes, the year-long job finished successfully!

**Comment by marshallward** [comment]
Gaea regression: https://gitlab.gfdl.noaa.gov/ogrp/SIS2/-/pipelines/19489 :heavy_check_mark:

**Comment by marshallward** [comment]
@kshedstrom Can I squash-merge this one?

**Comment by kshedstrom** [comment]
Yes, of course.

**Pull request merged**